samples/drawing/drawing.cpp

```diff
@@ -2130,14 +2136,40 @@ void MyFrame::OnSave(wxCommandEvent& WXUNUSED(event))
 #if wxUSE_LIBPNG
                      wxT("PNG image (*.png)|*.png;*.PNG|")
 #endif
+#if wxUSE_SVG
+                     wxT("SVG image (*.svg)|*.svg;*.SVG|")
+#endif
                      wxT("Bitmap image (*.bmp)|*.bmp;*.BMP"),
                      wxFD_SAVE | wxFD_OVERWRITE_PROMPT);
     if (dlg.ShowModal() == wxID_OK)
     {
-        wxBitmap bmp(500, 800);
-        wxMemoryDC mdc(bmp);
-        m_canvas->Draw(mdc);
-        bmp.ConvertToImage().SaveFile(dlg.GetPath());
+        int width, height;
+        m_canvas->GetVirtualSize(&width, &height);
+#if wxUSE_SVG
+        wxString ext = dlg.GetPath().AfterLast('.').Lower();
+        if (ext.Lower() == wxT("svg"))
+        {
+#if wxUSE_GRAPHICS_CONTEXT
+            wxGraphicsRenderer* tempRenderer = m_canvas->GetRenderer();
+            m_canvas->UseGraphicRenderer(NULL);
+#endif
+            wxSVGFileDC svgdc(dlg.GetPath(), width, height);
+            wxSVGBitmapHandler* bitmapHandler = new wxSVGBitmapEmbedHandler();
```

Comment:
Do we need to create this on the heap? Also, it looks like `svgdc` is left with a dangling pointer currently (which it hopefully doesn't use, but still...). I'd create the bitmap handler before `svgdc` on the stack, so it's automatically destroyed after it.